cruise.umple/src/GrammarParsing.ump

```diff
@@ -19,9 +104,19 @@ class RuleBasedParser
   
   public static HashMap<String,ChoiceRule> choicerules = new HashMap<String,ChoiceRule>();
   private static HashMap<String,String> todeclare = new HashMap<String,String>();
+  
+  // List of actions for the parser to perform. This has to be static because of choicerules and todeclare,
+  // but hopefully someday this will change.
+  private static HashMap<String,ParserAction> actionedTokens = new HashMap<String,ParserAction>();
```

Comment:
Oh, use a `HashMap`, but store it as a `Map` since the use of the `Map` doesn't care that it's a hash, or link, or anything for that matter. It's an implementation detail that bleeds into poor variable types. 

I can't remember the name of the rule, but you always want to *use* the most specific and necessary type but *store* it as the highest general type that matches. It was in Effective Java as some number that I can't remember off hand.